### ISSUE TITLE
Fails to create the first PR (renovate/configure) - Gitea API fail?

### ISSUE THREAD

**Issue by Adito5393** [renovatebot/renovate#10264]
**How are you running Renovate?**

- [ ] WhiteSource Renovate hosted app on github.com
- [x] Self hosted

If using the hosted app, please skip to the next section.
Otherwise, if self-hosted, please complete the following:

Please select which platform you are using:

- [ ] Azure DevOps (dev.azure.com)
- [ ] Azure DevOps Server
- [ ] Bitbucket Cloud (bitbucket.org)
- [ ] Bitbucket Server
- [x] Gitea 1.14.2
- [ ] github.com
- [ ] GitHub Enterprise Server
- [ ] gitlab.com
- [ ] GitLab self-hosted

Renovate version: 25.34.3

**Describe the bug**

The PR is not created. And I don't know how to debug it furthermore 😞 

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

```
DEBUG: Creating pull request: Configure Renovate (renovate/configure => master) (repository=Adi/opc-ua-client)
ERROR: Repository has unknown error (repository=Adi/opc-ua-client)
       "err": {
         "name": "HTTPError",
         "timings": {
           "start": 1622566650249,
           "socket": 1622566650249,
           "lookup": 1622566650250,
           "connect": 1622566650250,
           "upload": 1622566650250,
           "response": 1622566650522,
           "end": 1622566650523,
           "phases": {
             "wait": 0,
             "dns": 1,
             "tcp": 0,
             "request": 0,
             "firstByte": 272,
             "download": 1,
             "total": 274
           }
         },
         "message": "Response code 500 (Internal Server Error)",
         "stack": "HTTPError: Response code 500 (Internal Server Error)\n    at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:117:42)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
         "options": {
           "headers": {
             "user-agent": "https://github.com/renovatebot/renovate",
             "authorization": "***********",
             "accept": "application/json",
             "content-type": "application/json",
             "content-length": "3849",
             "accept-encoding": "gzip, deflate, br"
           },
           "url": "http://PRIVATE_URL/api/v1/repos/Adi/opc-ua-client/pulls",
           "username": "",
           "password": "",
           "method": "POST",
           "http2": false
         },
         "response": {
           "statusCode": 500,
           "statusMessage": "Internal Server Error",
           "body": {"message": "", "url": "http://PRIVATE_URL/api/swagger"},
           "headers": {
             "content-length": "63",
             "content-type": "application/json;charset=utf-8",
             "date": "Tue, 01 Jun 2021 16:57:30 GMT",
             "set-cookie": [
               "i_like_gitea=99336d1487ae4135; Path=/; HttpOnly",
               "_csrf=HOzJH5BBm4Zq7GPG1v2gAM9a3gg6MTYyMjU2NjY1MDI1MDg0ODMwOA; Path=/; Expires=Wed, 02 Jun 2021 16:57:30 GMT; HttpOnly; SameSite=Lax"
             ],
             "x-content-type-options": "nosniff",
             "x-frame-options": "SAMEORIGIN",
             "connection": "close"
           },
           "httpVersion": "1.1"
         }
       }
```

</details>

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [x] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

Is it normal for the header call that makes the POST for the new Pull request to contain the `"user-agent": "https://github.com/renovatebot/renovate"`?

The Gitea server is hosted internally only and available via HTTP! Not beeing HTTPS has something to do with the error?

**Comment by rarkins** [comment]
I think you need to check the Gitea logs for why it generated the 500 error.

**Comment by Adito5393** [comment]
@rarkins Thanks for the reply!

How/where can I see the POST body that `renovate` generated? 

It should be similar to the [gitea API example](https://try.gitea.io/api/swagger#/repository/repoCreatePullRequest):

```
{
  "assignee": "string",
  "assignees": [
    "string"
  ],
  "base": "string",
  "body": "string",
  "due_date": "2021-06-01T17:48:11.911Z",
  "head": "string",
  "labels": [
    0
  ],
  "milestone": 0,
  "title": "string"
}
```

Could the error be from an API incompatibility? or maybe I am missing some token? 

Running the same config.json file in `dryrun` does not give any errors (as expected).

**Comment by rarkins** [comment]
You can trace the source code that's used from here: https://github.com/renovatebot/renovate/blob/f5ae2d5db17039698900d24fe0dc8fefd394fb67/lib/platform/gitea/index.ts#L492-L498

**Comment by Adito5393** [comment]
Thanks, I will look into it and maybe run it with the log level set to trace (lots lots of printouts).

Meanwhile, can you have a quick look over the `config.json` I pass via the Drone CI:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "platform": "gitea",
  "endpoint": "http://GITEA-PRIVATE-URL/api/v1",
  "repositories": ["Adi/opc-ua-client"],
  "hostRules": [
    {
      "hostType": "maven",
      "matchHost": "http://PUBLIC-NEXUS-URL/"
    }
  ],
  "packageRules": [
    {
      "matchManagers":["maven"],
      "registryUrls": [
        "http://PUBLIC-NEXUS-URL/repository/maven-public/"
      ]
    }
  ]
}
```

With the following 2 vars supplied via CI secrets: `RENOVATE_TOKEN` and `GITHUB_COM_TOKEN`.

Is it correctly configurated for the Gitea platform?

**Comment by rarkins** [comment]
Config seems OK to me

**Comment by viceice** [comment]
Host rule is missing auth config, but it's not related to gitea.

I'm using gitea too and have no issues. So it must be related to your gitea server configuration.

So best steps are running renovate manually and view server logs in parallel.

**Comment by Adito5393** [comment]
Docker logs of gitea only says the same:

```log
2021/06/02 15:21:44 Started POST /Adi/opc-ua-client.git/git-upload-pack for 172.19.0.3:50406
2021/06/02 15:21:44 Completed POST /Adi/opc-ua-client.git/git-upload-pack 200 OK in 216.480652ms
2021/06/02 15:21:44 Started POST /Adi/opc-ua-client.git/git-upload-pack for 172.19.0.3:50406
2021/06/02 15:21:45 Completed POST /Adi/opc-ua-client.git/git-upload-pack 200 OK in 119.316058ms
2021/06/02 15:21:55 Started POST /api/v1/repos/Adi/opc-ua-client/pulls for 172.19.0.3:50406
2021/06/02 15:21:57 Completed POST /api/v1/repos/Adi/opc-ua-client/pulls 500 Internal Server Error in 1.552137546s
```

I will look into checking the logs with a different log level.

@viceice Did you have to modify something to the `app.ini` file of Gitea?
Maybe you set different: `PULL_REQUEST_QUEUE_LENGTH` or `PULL_REQUEST_PUSH_MESSAGE`? ([app.example.ini](https://github.com/go-gitea/gitea/blob/main/custom/conf/app.example.ini))

**Comment by viceice** [comment]
No, i don't changed those settings. I'm using latest docker hub image with the official helmchart.

**Comment by Adito5393** [comment]
@viceice I'm also using the latest Gitea stable version 1.14.2. I did try to run my renovate instance against [try.gitea](https://try.gitea.io/) and the PR is created without any issue on a private repo I have created there. Do you run your Gitea server on HTTP or HTTPS?
 
@rarkins I have updated renovate to latest Docker image and now I get more info on the error. Thanks! 
Any idea how to fix this error?

```log
 "body": {
             "message": "NewPullRequest: too many errors attempting to insert the new issue. Last error was: Error 1366: Incorrect string value: '\\xF0\\x9F\\x9A\\xA6 T...' for column 'content' at row 1",
             "url": "http://2git.powerspex.local/api/swagger"
           },
```

<details><summary>Click me to see the whole Error msg</summary>

```log
DEBUG: Creating pull request: Configure Renovate (renovate/configure => master) (repository=Adi/opc-ua-client)
ERROR: Repository has unknown error (repository=Adi/opc-ua-client)
       "err": {
         "name": "HTTPError",
         "timings": {
           "start": 1622708783256,
           "socket": 1622708783256,
           "lookup": 1622708783256,
           "connect": 1622708783256,
           "upload": 1622708783257,
           "response": 1622708784146,
           "end": 1622708784147,
           "phases": {
             "wait": 0,
             "dns": 0,
             "tcp": 0,
             "request": 1,
             "firstByte": 889,
             "download": 1,
             "total": 891
           }
         },
         "message": "Response code 500 (Internal Server Error)",
         "stack": "HTTPError: Response code 500 (Internal Server Error)\n    at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:117:42)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
         "options": {
           "headers": {
             "user-agent": "https://github.com/renovatebot/renovate",
             "authorization": "***********",
             "accept": "application/json",
             "content-type": "application/json",
             "content-length": "3849",
             "accept-encoding": "gzip, deflate, br"
           },
           "url": "http://2git.powerspex.local/api/v1/repos/Adi/opc-ua-client/pulls",
           "username": "",
           "password": "",
           "method": "POST",
           "http2": false
         },
         "response": {
           "statusCode": 500,
           "statusMessage": "Internal Server Error",
           "body": {
             "message": "NewPullRequest: too many errors attempting to insert the new issue. Last error was: Error 1366: Incorrect string value: '\\xF0\\x9F\\x9A\\xA6 T...' for column 'content' at row 1",
             "url": "http://2git.powerspex.local/api/swagger"
           },
           "headers": {
             "content-length": "240",
             "content-type": "application/json;charset=utf-8",
             "date": "Thu, 03 Jun 2021 08:26:24 GMT",
             "set-cookie": [
               "i_like_gitea=a822b3c0f70a8889; Path=/; HttpOnly",
               "_csrf=1GjONlTvRGKyMNklU6lndGkSgiM6MTYyMjcwODc4MzI1NzU0NTIzOA; Path=/; Expires=Fri, 04 Jun 2021 08:26:23 GMT; HttpOnly; SameSite=Lax"
             ],
             "x-content-type-options": "nosniff",
             "x-frame-options": "SAMEORIGIN",
             "connection": "close"
           },
           "httpVersion": "1.1"
         }
       }
```

</details>

**Comment by viceice** [comment]
It running behind traefik ingress, which is doing the https to http stuff.
I'm using PostgreSQL as database.

**Comment by Adito5393** [comment]
Any idea how to find out what DB is try.gitea.io is using? 
I'm using mysql DB 😞

**Comment by rarkins** [comment]
Could be a emoji-related?

**Comment by viceice** [comment]
Yes, i think so. MySQL has issues storing utf-8 two byte chars. Like the issue in bitbucket server with MySQL.

 So this is not a renovate bug.  You've multiple options:

1. Switch MySQL database to be Unicode compatible.
2. Use an other database like PostgreSQL
3. Disable Unicode emojis in renovate (this would not solve manual added emojis to issues. Pr's.)

https://docs.renovatebot.com/configuration-options/#unicodeemoji
